reframe/core/meta.py

```diff
@@ -58,7 +58,12 @@ def __init__(cls, name, bases, namespace, **kwargs):
         super().__init__(name, bases, namespace, **kwargs)
 
         # Create a set with the attribute names already in use.
-        used_attribute_names = set(dir(cls)) - set(cls.__dict__)
+        cls._rfm_dir = set()
+        for base in bases:
+            if hasattr(base, '_rfm_dir'):
+                cls._rfm_dir.update(base._rfm_dir)
+
+        used_attribute_names = copy.deepcopy(cls._rfm_dir)
```

Comment:
This should be enough, shouldn't be? `_rfm_dir` stores only strings, which are immutable, so I wouldn't go through the machinery of `copy.deepcopy()`.

```suggestion
        used_attribute_names = set(cls._rfm_dir)
```

Reply:
Yes, good point. I'll change that.